### PR TITLE
Fix WebSocket-triggered table flicker with Alpine morph

### PR DESCRIPTION
**Pull request by vpetersson**
## Summary

- Switch all 6 table templates from `hx-swap="outerHTML"` to `hx-swap="morph"` so `Alpine.morph()` preserves reactive state (search, filters, sort, pagination) instead of tearing down and rebuilding the DOM on every WebSocket event
- Add 500ms debounce to all WebSocket handlers so rapid events (e.g. bulk uploads) coalesce into a single HTMX refresh
- Move `json_script` data blocks inside the container div (4 core tables) and add `htmx:afterSettle` listener to re-read fresh data after morph, with `currentPage` clamping for deletion edge cases

## Test plan

- [ ] Rapid WebSocket events: upload/create multiple components quickly — table should update smoothly without flashing
- [ ] State preservation: apply a search filter, go to page 2, then trigger a WS event — search and page should be preserved
- [ ] Pagination accuracy: create a component while viewing the table — "Showing X to Y of Z entries" should update Z without resetting page
- [ ] Empty → non-empty: create the first component while on empty state — table should appear
- [ ] Page clamping: delete items while on the last page — currentPage should adjust down

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: sbomify/apps/sboms/templates/sboms/sboms_table.html.j2 [hunk 1]
```diff
@@ -5,8 +5,8 @@
      x-data
      hx-get="{% url 'sboms:sboms_table' component_id %}"
      hx-trigger="refresh-sboms from:body"
-     hx-swap="outerHTML"
-     @ws:message.window="if (['sbom_uploaded', 'sbom_deleted'].includes($event.detail.type) && $event.detail.component_id === '{{ component_id }}') { document.body.dispatchEvent(new CustomEvent('refresh-sboms')); }">
+     hx-swap="morph"
```
[the remark this answers is not in the document]
Comment (vpetersson) [reply sbomify/apps/sboms/templates/sboms/sboms_table.html.j2]: Fixed in 21e2ed00. Added `json_script` data block + `init()` with `htmx:afterSettle` listener to re-read fresh data after morph, plus `destroy()` for cleanup.

File: sbomify/apps/core/templates/core/products_table.html.j2 [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
...); }, get paginatedData() { const start = (this.currentPage - 1) * this.perPage; return this.sortedData.slice(start, start + this.perPage); }, get totalPages() { return Math.ceil(this.filteredData.length / this.perPage); }, get startIndex() { return this.filteredData.length > 0 ? (this.currentPage - 1) * this.perPage + 1 : 0; }, get endIndex() { return Math.min(this.currentPage * this.perPage, this.filteredData.length); }, sort(column) { if (this.sortColumn === column) { this.sortDirection = this.sortDirection === 'asc' ? 'desc' : 'asc'; } else { this.sortColumn = column; this.sortDirection = 'asc'; } }, goToPage(page) { if (page >= 1 && page <= this.totalPages) { this.currentPage = page; } }, get visiblePages() { const pages = []; const total = this.totalPages; const current = this.currentPage; if (total <= 7) { for (let i = 1; i <= total; i++) pages.push(i); } else { pages.push(1); if (current > 3) pages.push('...'); for (let i = Math.max(2, current - 1); i <= Math.min(total - 1, current + 1); i++) { pages.push(i); } if (current < total - 2) pages.push('...'); pages.push(total); } return pages; }, getProjectColor(idx) { return 'bg-border/20 text-text-muted border-border/40'; } }"
```
[the remark this answers is not in the document]
Comment (vpetersson) [reply sbomify/apps/core/templates/core/products_table.html.j2]: Same as above — with morph, `init()` runs once per page load. No duplicate listener risk.
